workers/unity/Assets/Fps/Scripts/GameLogic/Hud/Mobile/StandardButton.cs

```diff
@@ -0,0 +1,192 @@
+﻿using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+[SelectionBase]
+public class StandardButton : MonoBehaviour, IPointerDownHandler, IPointerUpHandler, IDragHandler
+{
+    [Tooltip("How long after pressing the button will register another press")]
+    public float CooldownTime;
+
+    [Tooltip("Whether a click dragged out of the button area ends the press or not")]
+    public bool RestrictPressToButtonArea;
+
+    [Tooltip("The hitbox used to detect if click is dragged out of the button area")]
+    public Graphic Hitbox;
+
+    public bool Togglable;
+
+    public bool IsPressed { get; private set; }
+
+    public delegate void ButtonEvent(PointerEventData eventData);
+
+    public ButtonEvent OnButtonDown;
+    public ButtonEvent OnButtonUp;
+    public ButtonEvent OnButtonDrag;
+
+    private float lastPressedTime;
+
+    private StandardButtonAnimator[] animators;
+
+    private void OnValidate()
+    {
+        CooldownTime = Mathf.Max(0, CooldownTime);
+        OnValidateCheckHitbox();
+    }
+
+    private void OnValidateCheckHitbox()
+    {
+        if (Hitbox != null)
+        {
+            return;
+        }
+
+        var allGraphics = GetComponentsInChildren<Graphic>();
+        foreach (var graphic in allGraphics)
+        {
+            if (!graphic.raycastTarget)
+            {
+                continue;
+            }
+
+            Hitbox = graphic;
+            Debug.LogWarning(
+                $"Automatically located and added Hitbox reference to button {gameObject.name}.\n" +
+                $"You may want to assign your own!\n");
+            break;
+        }
+
+        Debug.LogWarning($"No hitbox found for button {gameObject.name}!\n");
+    }
+
+    private void Awake()
+    {
+        animators = GetComponentsInChildren<StandardButtonAnimator>();
+    }
+
+    // Presently it is assumed the player is dead if the UI is disabled, so reset everything
+    private void OnDisable()
+    {
+        SetPressedTo(false);
+    }
+
+
+    public void SetPressedTo(bool pressed)
+    {
+        // Not terribly nice passing in null PointerEventData refs here,
+        // but unsure of a better solution for now
+        if (pressed)
+        {
+            Press(null);
```

Comment:
cooould do...but since Press() in turn calls an event that passes along the PointerEventData, maybe there's a benefit to making it explicit that you're passing around null arguments, and so can be sure to handle them, rather than happening under the hood?